### ISSUE TITLE
Repair `atd_kits_sig_status_to_socrata`; notify appropriate parties of the unreachable MySQL server

### ISSUE THREAD

**Issue by frankhereford** [cityofaustin/atd-data-tech#9004]
## TLDR
~KITS integration is down.~ We learned that it was the KITS system itself that was down. It was a planned outage KH needed to perform an update. 

## Note
This is my first exposure to these scripts. While I'm comfortable with dealing with what they do at a technical level (mysql connections, etc), I am unfamiliar with the task they are meant to execute and who the stakeholders are. Any information and guidance about the degree of urgency this issue has or who I may need to speak to about that remote server is very welcome!

## Details
@chiaberry brought it to my attention that an airflow DAG was failing. The ETL in question [is this one](https://github.com/cityofaustin/atd-kits/blob/production/atd-kits/signal_status_publisher.py) in the `atd-kits` repo. The failure is [found here](https://austininnovation.slack.com/archives/C013G4RNJ01/p1650994984300209?thread_ts=1650994923.293779&cid=C013G4RNJ01), and indicates an issue with the connection to the mysql server running on `10.66.20.157` which provides a mysql database named `KITSDB`. I have spun up a docker container with a mysql client, and I am unable to connect to the server with the following command: `mysql -u ATD_FME_SERVICE -p***** -h 10.66.20.157 -D KITSDB`.

@dianamartin kindly referred us to some folks: "Brian C. if it’s server related. But maybe Lance and Allyson know about it (MMC).", and I have reached out to Brian on teams. I stated:

>[1:08 PM] Hereford, Frank
Brian, how are you doing - I hope well. My name is Frank Hereford and I'm part of the DTS team under Amenity A. (previously John C..)
> One of our integration scripts which connects to a mysql database on 10.66.20.157 to use the KITS_DATABASE there has started to fail with an error. Initially, it through an exception and suggested looking at the server lots, but now I am unable to make a connection to the server at all with a mysql client. 
> Is this something you can help me with, or point me in the right direction?

@chiaberry has disabled the DAG for now, and I will update this issue with more information as I find it.

**Comment by frankhereford** [comment]
Brian responded with:
> You can try reaching out to Chad Didion. You can CC me on your email or message so he knows that I am aware.

I am going to follow his suggestion and reach out to Chad on teams.

**Comment by frankhereford** [comment]
@mateoclarke gave some more information - thank you!

He indicated that he believed that the KITS database is managed / operated by Kimley Horn. 

I will reach out to her via email if I don't hear back from Chad in response to the teams message I sent after Brian suggested that message him.

**Comment by amenity** [comment]
Thanks, @frankhereford @chiaberry @mateoclarke . I'm pretty sure that while we maintain these ETLs, the KITS platform is outside our purview. Brian C. can confirm. I can email Brian to confirm that distinction and hopefully close out the issue on our end.

**Comment by frankhereford** [comment]
@amenity wrote:
> Thanks, @frankhereford @chiaberry @mateoclarke . I'm pretty sure that while we maintain these ETLs, the KITS platform is outside our purview. I will email Brian to confirm that distinction and hopefully close out the issue on our end.

Thanks Amenity - that's perfect. Thank you for doing that. I've set a reminder in slack to turn the DAG back on in the morning. @chiaberry kindly turned it off so we wouldn't get notified every five minutes in the airflow channel.

**Comment by amenity** [comment]
There was a KITS update yesterday that caused this outage, so closing on our end. I learned a lot though! 😅 

Good info from Lance via email: 
> Kimley-Horn manages the KITS services on-premise with assistance from CTM who helps manage the COA servers upon which KITS runs.
DTS has process(es?) that query data from KITS to publish to Socrata which we see as a DTS process, but one that Kimley-Horn assists with from the KITS platform side as need😅

#### KIT-y things I had never looked at before this incident

**Related repos**
- https://github.com/cityofaustin/atd-kits
- https://github.com/cityofaustin/atd-utils-kits 

**Current references to KITS in our docs**
https://atd-dts.gitbook.io/open-data/datasets/radar-traffic-counts
https://atd-dts.gitbook.io/atd-knack-operations/data-tracker/cameras

**Existing issues**
- https://github.com/cityofaustin/atd-data-tech/issues/1600
- https://github.com/cityofaustin/atd-data-tech/issues/8328

**Comment by mateoclarke** [comment]
I don't know if this is ready to close. The DAG isn't yet functional.

**Comment by amenity** [comment]
Aha, thanks @mateoclarke. Catching up on Slack now. I'll hand closing duty over to y'all. ;)

**Comment by mateoclarke** [comment]
Adding myself as assigned to this issue. I've shared with LB our current "permissions denied" error and suggested the need for enhanced permissions for our account to access the KITS DB.

**Comment by frankhereford** [comment]
Chad at CTM responded to my teams message from Tuesday today via email:

> the vendors at kimley horn have also said the backups on the sql server are no longer working..    does the script you are having issues with have anything to do with backups.. or what is the scripts function.?.   im thinking maybe antivirus software but wanted some more info

**Comment by frankhereford** [comment]
Cole R. at Kimley-Horn wrote the group asking that we try to access the mysql server again after they found a "SME SQL role that was changed" which they reverted. The DAG was turned back up and is processing data once again. Thanks everyone!
